warehouse/tuf/repository.py

```diff
@@ -0,0 +1,490 @@
+        """
+        Loads latest version of metadata for rolename using configured storage backend.
+
+        NOTE: The storage backend is expected to translate rolenames to filenames.
+
+        Returns:
+            Role metadata
+            ``tuf.api.metadata.Metadata``
+        """
+        return Metadata.from_file(rolename, None, self.storage_backend)
+
+    def delegate_targets_roles(
+        self,
+        payload: Dict[str, List[RolesPayload]],
+    ) -> Metadata[Snapshot]:
+        """
+        Performs targets delegation for delegator-to-delegates items in passed payload.
+
+        Creates new basic delegate metadata, configures delegation in delegator metadata
+        and bumps its version, and updates snapshot metadata accordingly.
+
+        Args:
+            payload: Dictionary of delegator role names as keys and lists of
+                    per-delegate info to populate the delegate metadata.
+
+        Raises:
+            FileExistsError: Delegate metadata already exists.
+
+        Returns:
+            Updated snapshot metadata
+            ``tuf.api.metadata.Metadata[Snapshot]``
+        """
+
+        snapshot_metadata = self.load_role(Snapshot.type)
+        for delegator, delegate_role_parameters in payload.items():
+            delegator_metadata = self.load_role(delegator)
+            snapshot_metadata = self._create_delegated_targets_roles(
+                delegator_metadata, snapshot_metadata, delegate_role_parameters
+            )
+            delegator_metadata = self.bump_role_version(
+                rolename=delegator,
+                role_metadata=delegator_metadata,
+                role_expires=delegator_metadata.signed.expires,
+                key_rolename=None,
+                store=True,
+            )
+            snapshot_metadata = self.snapshot_update_meta(
+                delegator, delegator_metadata.signed.version, snapshot_metadata
+            )
+
+        return snapshot_metadata
+
+    def bump_role_version(
+        self,
+        rolename: str,
+        role_metadata: Metadata,
+        role_expires: datetime,
+        key_rolename: Optional[str] = None,
+        store: Optional[bool] = False,
+    ) -> Metadata:
+        """
+        Bumps metadata version by one and assigns new expiration date for passed role.
+
+        Args:
+            rolename: Used to associate signing key and (optionally) store metadata.
+            role_metadata: Role metadata to be bumped.
+            role_expires: New role expiration date.
+            key_rolename: Used to associate a signing key by a name other than rolename.
+            store: Indicates whether metadata should be written to storage.
+
+        Returns:
+            Updated metadata
+            ``tuf.api.metadata.Metadata``
+        """
+        if key_rolename:
+            key_rolename = key_rolename
+        else:
+            key_rolename = rolename
+        role_metadata.signed.expires = role_expires
```

Comment:
It probably does not hurt, other than adding a bit of cruft. The expiration date is determined by a [per-role configured interval](https://github.com/pypa/warehouse/pull/10870/files#diff-d97469a400f4dc4aeedfb7bf4447cccf591ad162e3a41fb4933763c25db77524R23-R34) and a [helper function](https://github.com/pypa/warehouse/pull/10870/files#diff-5951c0db92c691317caa1d202936072a1607ff602a28f7896efa233ea4ffebb4R210), both controlled by us. Maybe instead of checking here we could add tests that the configured value is greater 0 and the helper indeed adds the value to now?